### ISSUE TITLE
STGCN model output not in range (0, 1)

### ISSUE THREAD

**Issue by MABatin** [open-mmlab/mmaction2#2444]
### Branch

0.x branch (0.x version, such as `v0.24.1`)

### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmaction2/issues) and [Discussions](https://github.com/open-mmlab/mmaction2/discussions) but cannot get the expected help.
- [X] I have read the [documentation](https://mmaction2.readthedocs.io/en/latest/) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version](https://github.com/open-mmlab/mmaction2).

### Environment

sys.platform: linux
Python: 3.8.10 (default, Mar 13 2023, 10:26:41) [GCC 9.4.0]
CUDA available: True
GPU 0: NVIDIA GeForce GTX 1080
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 11.3, V11.3.109
GCC: x86_64-linux-gnu-gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
PyTorch: 1.12.1+cu113
PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.3
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.2.1
    - Built with CuDNN 8.3.2
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.3.2, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -fabi-version=11 -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.12.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

TorchVision: 0.13.1+cu113
OpenCV: 4.5.5
MMCV: 1.6.2
MMCV Compiler: GCC 9.4
MMCV CUDA Compiler: 11.3
MMAction2: 0.24.1+dbb8f06

### Describe the bug

After training STGCN model on a custom dataset with 3 classes I use the test.py script to produce result.pkl file. But the output of the pickle file i.e. scores of each categories are as following:
```[array([112.8751  , -26.707651, -88.75476 ], dtype=float32), array([ 72.66257 , -17.159836, -57.156334], dtype=float32), array([ 166.94261 ,  -39.301228, -131.46951 ], dtype=float32), array([ 93.48183, -22.08731, -73.56408], dtype=float32), array([ 131.81519 ,  -30.876627, -103.92588 ], dtype=float32), array([ 139.42256 ,  -32.902008, -109.720985], dtype=float32), array([ 98.84652 , -23.399982, -77.74518 ], dtype=float32), array([ 133.25919 ,  -31.424435, -104.89922 ], dtype=float32), array([ 151.73079 ,  -35.87758 , -119.358505], dtype=float32), array([ 153.59714 ,  -36.27394 , -120.875824], dtype=float32), array([112.95138 , -26.675314, -88.88019 ], dtype=float32), array([113.21502 , -26.727423, -89.0889  ], dtype=float32), array([116.69247 , -27.532963, -91.83771 ], dtype=float32), array([ 99.64529 , -23.532692, -78.400604], dtype=float32), array([ 210.69145,  -49.91717, -165.69133], dtype=float32), array([ 129.7979  ,  -30.629822, -102.145256], dtype=float32), array([ 162.18231 ,  -38.212658, -127.69174 ], dtype=float32), array([ 164.18549,  -38.70068, -129.26273], dtype=float32), array([117.79175 , -27.789833, -92.70568 ], dtype=float32), array([ 131.45576 ,  -31.005884, -103.47404 ], dtype=float32), array([104.188515, -24.597654, -81.98135 ], dtype=float32), array([120.482155, -28.423132, -94.8247  ], dtype=float32), array([ 130.36198 ,  -30.715853, -102.63647 ], dtype=float32)]```

As can be seen, only the first class has positive score and the rest negative and they are not in (0, 1) range. 

Again, after training the model with ```test_cfg=dict(average_clips='prob')``` in the config file, I get the following output scores:
```[array([ 0.2754167 ,  0.09092851, -0.36069983], dtype=float32), array([ 0.17698571,  0.19334312, -0.35601163], dtype=float32), array([ 1.2115313 , -0.12632683, -1.107461  ], dtype=float32), array([ 0.26693752,  0.13226186, -0.39108557], dtype=float32), array([ 0.56038684,  0.00942538, -0.57307506], dtype=float32), array([ 0.663207  ,  0.00464722, -0.67314565], dtype=float32), array([ 0.22469527,  0.13039005, -0.34562767], dtype=float32), array([ 0.35040203,  0.04900473, -0.39689374], dtype=float32), array([ 1.0409738 , -0.07687278, -0.98131347], dtype=float32), array([ 0.24904719,  0.11606301, -0.35665148], dtype=float32), array([ 0.40709352,  0.06221253, -0.46731228], dtype=float32), array([ 0.15501155,  0.17291827, -0.31307453], dtype=float32), array([ 0.47875977,  0.05100122, -0.5291404 ], dtype=float32), array([ 0.20392734,  0.13564049, -0.32894725], dtype=float32), array([ 1.0126173 , -0.21665688, -0.8252386 ], dtype=float32), array([ 0.43139163,  0.0650667 , -0.4941728 ], dtype=float32), array([ 0.5577469,  0.0131937, -0.5731776], dtype=float32), array([ 0.88311744, -0.0453369 , -0.8497149 ], dtype=float32), array([ 0.5011562 ,  0.04486234, -0.5460944 ], dtype=float32), array([ 0.4188035 ,  0.06616421, -0.48266578], dtype=float32), array([ 0.14158207,  0.16268967, -0.29019558], dtype=float32), array([ 0.5862252 ,  0.02307083, -0.6119745 ], dtype=float32), array([ 0.37879053,  0.08160764, -0.4563641 ], dtype=float32)]```

Is this normal? Or am I missing something crucial?

### Reproduces the problem - code sample

My config file is as follows:

```
model = dict(
    type='SkeletonGCN',
    backbone=dict(
        type='STGCN',
        in_channels=3,
        edge_importance_weighting=True,
        graph_cfg=dict(layout='coco', strategy='spatial')),
    cls_head=dict(
        type='STGCNHead',
        num_classes=3,
        in_channels=256,
        loss_cls=dict(type='CrossEntropyLoss')),
    train_cfg=None,
    test_cfg=dict(average_clips='prob'))

dataset_type = 'PoseDataset'
ann_file_train = '/home/portia/portia_backup/fall/mmaction2/data/ntu-fall/ntu-fall_xsub_train.pkl'
ann_file_val = '/home/portia/portia_backup/fall/mmaction2/data/ntu-fall/ntu-fall_xsub_val.pkl'
train_pipeline = [
    dict(type='PaddingWithLoop', clip_len=300),
    dict(type='PoseDecode'),
    dict(type='FormatGCNInput', input_format='NCTVM'),
    dict(type='PoseNormalize'),
    dict(type='Collect', keys=['keypoint', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['keypoint'])
]
val_pipeline = [
    dict(type='PaddingWithLoop', clip_len=300),
    dict(type='PoseDecode'),
    dict(type='FormatGCNInput', input_format='NCTVM'),
    dict(type='PoseNormalize'),
    dict(type='Collect', keys=['keypoint', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['keypoint'])
]
test_pipeline = [
    dict(type='PaddingWithLoop', clip_len=300),
    dict(type='PoseDecode'),
    dict(type='FormatGCNInput', input_format='NCTVM'),
    dict(type='PoseNormalize'),
    dict(type='Collect', keys=['keypoint', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['keypoint'])
]
data = dict(
    videos_per_gpu=16,
    workers_per_gpu=2,
    test_dataloader=dict(videos_per_gpu=1),
    train=dict(
        type=dataset_type,
        ann_file=ann_file_train,
        data_prefix='',
        pipeline=train_pipeline),
    val=dict(
        type=dataset_type,
        ann_file=ann_file_val,
        data_prefix='',
        pipeline=val_pipeline),
    test=dict(
        type=dataset_type,
        ann_file=ann_file_val,
        data_prefix='',
        pipeline=test_pipeline))

# optimizer
optimizer = dict(
    type='SGD', lr=0.1, momentum=0.9, weight_decay=0.0001, nesterov=True)
optimizer_config = dict(grad_clip=None)
# learning policy
lr_config = dict(policy='step', step=[10, 50])
total_epochs = 80
checkpoint_config = dict(interval=5)
evaluation = dict(interval=5, metrics=['top_k_accuracy'])
log_config = dict(interval=100, hooks=[dict(type='TextLoggerHook')])

# runtime settings
dist_params = dict(backend='nccl')
log_level = 'INFO'
work_dir = './work_dirs/stgcn_80e_ntu-fall_xsub_keypoint/'
load_from = None
resume_from = None
workflow = [('train', 1)]
gpu_ids = [0]
```

### Reproduces the problem - command or script

```
python train.py ../configs/skeleton/stgcn/stgcn_80e_ntu-fall_xsub_keypoint.py \
    --validate --seed 0 --deterministic
```

### Additional information

1. I expected the result to be in range (0, 1)
2. I used a custom dataset with 3 classes

**Comment by MABatin** [comment]
Also model top1 accuracy graph looks like following

![image](https://user-images.githubusercontent.com/56408528/235656281-c8168e89-0cfb-4138-b155-59f321eb6662.png)

Is this because my dataset has small number of samples (89 to be exact)?

**Comment by Dai-Wenxun** [comment]
I think it's normal because the dumped `results.pkl`  contains the raw classification scores w/o softmax. Maybe you can apply a softmax to each item in your `results.pkl` to make them in range(0, 1).
